web/src/upload.js

```diff
@@ -318,12 +318,18 @@ export function setup_upload(config) {
         const split_url = url.split("/");
         const filename = split_url.at(-1);
         const filename_url = "[" + filename + "](" + url + ")";
-        compose_ui.replace_syntax(
+        const text_area = get_item("textarea", config);
+        const replacement_successful = compose_ui.replace_syntax(
             get_translated_status(file),
             filename_url,
-            get_item("textarea", config),
+            text_area,
         );
-        compose_ui.autosize_textarea(get_item("textarea", config));
+        if (!replacement_successful) {
+            const new_content = `${text_area.val()} ${filename_url}`;
+            text_area.val(new_content);
+        }
```

Comment:
We should never directly edit the content of a text area like this; we need to use one of the functions from `compose_ui` to adjust the compose box content without breaking browser undo. Probably just a call to `insert_syntax_and_focus` with the right parameters would suffice.

Reply:
Ah great, thanks! Just pushed a change using that instead.